tests/test_context.py

```diff
@@ -0,0 +1,46 @@
+# Source pytest module
+# Copyright (C) 2020  Ngô Ngọc Đức Huy
+#
+# This file is part of palace.
+#
+# palace is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published
+# by the Free Software Foundation, either version 3 of the License,
+# or (at your option) any later version.
+#
+# palace is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with palace.  If not, see <https://www.gnu.org/licenses/>.
+
+"""This pytest module tries to test the correctness of the class Context."""
+
+from palace import Context
+
+
+def test_with_context(device):
```

Comment:
I think this can be removed now.  Check whether it contribute to the coverage.  Also if this is removed, please rename the other test to avoid confusion.

Reply:
What do you want  the other test be named to? I think `test_nested_context` is descriptive enough. Also I think keeping this test is convenient because if it passes and some other tests don't, then it's not because there's some problem with using `with context`.

Reply:
> I think keeping this test is convenient because if it passes and some other tests don't

That's fine to me, the whole idea of security if redundancy anyway.

> I think `test_nested_context` is descriptive enough.

Maybe changing it to `test_nested_context_manager`, since that's what's nested (I guess it confused you too when you name it :wink:).